exercise.wwwapi/Endpoints/ProductEndpoints.cs

```diff
@@ -0,0 +1,144 @@
+using Microsoft.OpenApi.Writers;
+using Microsoft.AspNetCore.Components.Forms;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using System.Net;
+using System;
+using Microsoft.AspNetCore.Mvc.Filters;
+using System.Net.Http;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Diagnostics;
+
+namespace exercise.wwwapi.Endpoints
+{
+
+    [ApiController]
+    public static class ProductEndpoints
+    {
+        public static void ConfigureProductEndpoints(this WebApplication app)
+        {
+
+            var group = app.MapGroup("products");
+            group.MapGet("/", GetAllProducts);
+            group.MapGet("/{id}", GetProduct);
+            group.MapPost("/", CreateProduct);
+            group.MapPut("/{id}", UpdateProduct);
+            group.MapDelete("/{id}", DeleteProduct);
+
+            app.UseExceptionHandler(c => c.Run(async context =>
+            {
+                var exception = context.Features
+                    .Get<IExceptionHandlerFeature>()
+                    ?.Error;
+                if (exception is not null)
+                {
+                    var response = new { error = exception.Message };
+                    context.Response.StatusCode = 400;
+
+                    await context.Response.WriteAsJsonAsync("Price must be an integer, something else was provided.");
+                }
+            }));
+
+        }
+
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public static IResult GetAllProducts(IProductRepository products, string? category)
+        {
+
+            List<string> cats = products.GetAllProducts(null).Select(x => x.Category).ToList();
+
+            if (category != null && !cats.Contains(category))
+            {
+                return TypedResults.NotFound("No products of the provided category were found.");
+            }
+
+            return TypedResults.Ok(products.GetAllProducts(category));
+        }
+
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public static IResult GetProduct(IProductRepository products, int id)
+        {
+            Product? prod = products.GetProduct(id);
+            if (prod == null)
+            {
+                return TypedResults.NotFound($"Product not found.");
+            }
+            return TypedResults.Ok(prod);
+        }
+
+      
+
+
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public static IResult CreateProduct([FromBody] ProductPostPayload newData,  IProductRepository products) 
+        {
+           
+            if (newData.Name == null || newData.Category == null || newData.Price == null) return TypedResults.BadRequest("All fields are required.");
```

Comment:
It is better to do model validation in the model, perhaps using the [Required] attribute